src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/AppointmentDelivery.kt

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity
+
+import com.vladmihalcea.hibernate.type.basic.PostgreSQLEnumType
+import org.hibernate.annotations.Type
+import org.hibernate.annotations.TypeDef
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.Id
+import javax.persistence.OneToOne
+import javax.persistence.PrimaryKeyJoinColumn
+import javax.validation.constraints.NotNull
+
+@Entity
+@TypeDef(name = "appointment_delivery_type", typeClass = PostgreSQLEnumType::class)
+data class AppointmentDelivery(
+  @Id
+  var appointmentId: UUID,
+  @Type(type = "appointment_delivery_type")
+  @Enumerated(EnumType.STRING)
+  @NotNull var appointmentDeliveryType: AppointmentDeliveryType,
```

Comment:
Can we introduce referential integrity validation, where we test the accompanying option against the detail field?

I.e. make it invalid to set it `VIDEO_CALL` without `videoCallUrl` etc etc.

Reply:
video call url is no longer required - we just set the type

I'd rather have that logic within the code since introducing these constraints makes it more complicated when it comes to updating/deleting (due to hibernate not understanding which order to perform operations) 

but if you feel strongly I will do it.

Reply:
yeah, service code is fine -- I couldn't find it though, is that in the other PR?